src/main/java/edu/kis/powp/jobs2d/drivers/InkUsageDriverDecorator.java

```diff
@@ -0,0 +1,110 @@
+package edu.kis.powp.jobs2d.drivers;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.observer.Publisher;
+
+import java.util.logging.Logger;
+
+public class InkUsageDriverDecorator implements Job2dDriver {
+    private int x0, y0;
+    private double inkLimit;
+    private double totalUsage;
+    private Logger logger = Logger.getLogger("global");
+    private Job2dDriver driver;
+    private Publisher pub;
+    private boolean isUnAvailable = false;
+    private double maxInkLimit;
+
+    public InkUsageDriverDecorator(Job2dDriver driver, double maxInkLimit) {
+        super();
+        this.driver = driver;
+        this.x0 = 0;
+        this.y0 = 0;
+        this.totalUsage = 0;
+        this.inkLimit = maxInkLimit;
+        this.maxInkLimit = this.inkLimit;
+        pub = new Publisher();
+        pub.addSubscriber(new NoInkObserver(this));
+    }
+
+    public InkUsageDriverDecorator(Job2dDriver driver, double inkLimit, double totalUsage, double maxInkLimit) {
+        super();
+        this.driver = driver;
+        this.x0 = 0;
+        this.y0 = 0;
+        this.totalUsage = totalUsage;
+        this.inkLimit = inkLimit;
+        this.maxInkLimit = maxInkLimit;
+        pub = new Publisher();
+        pub.addSubscriber(new NoInkObserver(this));
+    }
+
+
+    @Override
+    public void setPosition(int x, int y) {
+        double wasted = inkCounter(this.x0, x, this.y0, y);
+        if((inkLimit-wasted) >= 0){
```

Comment:
setPosition does not affect INK (operation should be executed unconditionally), but it should affect the totalUsage.

Reply:
When it comes to this point we assumed that totalUsage is measure for whole Ink used by this machine (not just the head) but anyway we adjusted it to Yours suggestion.

Reply:
I agree with you: I didn't want to add: totalUsage += wasted.
 I meant that 'if' statement seemed to be unnecessary.